services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/allu/CableReportServiceAlluITests.kt

```diff
@@ -80,15 +80,66 @@ class CableReportServiceAlluITests {
 
         assertThrows<WebClientResponseException.BadRequest> { service.create(getTestApplication()) }
 
-        val loginRequest = mockWebServer.takeRequest()
-        assertThat(loginRequest.method).isEqualTo("POST")
-        assertThat(loginRequest.path).isEqualTo("/v2/login")
-        assertThat(loginRequest.getHeader("Authorization")).isNull()
-
+        assertThat(mockWebServer.takeRequest()).isValidLoginRequest()
         val createRequest = mockWebServer.takeRequest()
         assertThat(createRequest.method).isEqualTo("POST")
         assertThat(createRequest.path).isEqualTo("/v2/cablereports")
-        assertThat(createRequest.getHeader("Authorization")).isEqualTo("Bearer " + stubbedBearer)
+        assertThat(createRequest.getHeader("Authorization")).isEqualTo("Bearer $stubbedBearer")
+    }
+
+    @Nested
+    inner class GetDecisionPdf {
+        @Test
+        fun `returns PDF file as bytes`() {
+            val stubbedBearer = addStubbedLoginResponse()
+            val pdfBytes = "/fi/hel/haitaton/hanke/decision/fake-decision.pdf".getResourceAsBytes()
+            val buffer = Buffer()
+            buffer.write(pdfBytes)
+            mockWebServer.enqueue(
+                MockResponse()
+                    .setResponseCode(200)
+                    .setHeader(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_PDF_VALUE)
+                    .setBody(buffer)
+            )
+
+            val response = service.getDecisionPdf(12)
+
+            assertThat(response).isEqualTo(pdfBytes)
+            assertThat(mockWebServer.takeRequest()).isValidLoginRequest()
+            val createRequest = mockWebServer.takeRequest()
+            assertThat(createRequest.method).isEqualTo("GET")
+            assertThat(createRequest.path).isEqualTo("/v2/cablereports/12/decision")
+            assertThat(createRequest.getHeader("Authorization")).isEqualTo("Bearer $stubbedBearer")
+        }
+
+        @Test
+        fun `throws ApplicationDecisionNotFoundException on 404`() {
+            addStubbedLoginResponse()
+            mockWebServer.enqueue(
+                MockResponse()
+                    .setResponseCode(404)
+                    .setHeader(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
+                    .setBody("Error message")
```

Comment:
This body could be a constant. Feel free to ignore.

Reply:
I don't think aliasing these would make this test more readable. I changed the very generic error messages to be slightly more descriptive.